### PR TITLE
[libcommhistory] Implement limit/offset for event models. Fixes MER#1291

### PR DESCRIPTION
**Pull request by matthewvogt**

### CODE REVIEW

File: src/callmodel.cpp [hunk 1]
```diff
@@ -1134,7 +1120,7 @@ bool CallModel::markAllRead()
     bool marked;
     marked = d->database()->markAsReadAll(Event::CallEvent);
     if (!marked) {
-        qWarning() << Q_FUNC_INFO << "Failed to delete events";
+        qWarning() << Q_FUNC_INFO << "Failed to mark events";
```
Comment (chriadam) [inline src/callmodel.cpp]: "... as read"

File: src/callmodel.cpp [hunk 2]
```diff
@@ -94,8 +94,6 @@ class CallModelPrivate : public EventModelPrivate
 
     void deleteFromModel( int id );
 
-    void deleteCallGroup( const Event &event, bool typed );
```
Comment (chriadam) [inline src/callmodel.cpp]: was this an unimplemented function prototype?

File: tests/ut_singlecontacteventmodel/singlecontacteventmodeltest.cpp [hunk 3]
```diff
@@ -140,4 +143,80 @@ void SingleContactEventModelTest::testGetEvents()
...
+    QVERIFY(model.getEvents(contactId));
+    QTRY_VERIFY(model.isReady());
+    QCOMPARE(model.rowCount(), 4);
+
+    QCOMPARE(model.event(model.index(0, 0)).id(), eventIds.at(0));
+    QCOMPARE(model.event(model.index(1, 0)).id(), eventIds.at(1));
+    QCOMPARE(model.event(model.index(2, 0)).id(), eventIds.at(2));
+    QCOMPARE(model.event(model.index(3, 0)).id(), eventIds.at(3));
+
+    model.setLimit(2);
```
Comment (chriadam) [inline tests/ut_singlecontacteventmodel/singlecontacteventmodeltest.cpp]: so, is this guaranteed to get exactly 2 from offset 0?

**Comment by chriadam** [comment]
Minor comments, one question, but LGTM.

File: src/callmodel.cpp (hunk 2 above)
Comment (matthewvogt) [inline src/callmodel.cpp]: Not sure if never implemented or previously removed, but it doesn't exist now.

File: tests/ut_singlecontacteventmodel/singlecontacteventmodeltest.cpp (hunk 3 above)
Comment (matthewvogt) [inline tests/ut_singlecontacteventmodel/singlecontacteventmodeltest.cpp]: Yes.

File: src/callmodel.cpp (hunk 1 above)
Comment (matthewvogt) [inline src/callmodel.cpp]: Fixed.

**Comment by chriadam** [comment]
LGTM

**Pull request merged**